assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -2318,7 +2318,7 @@ public String getGradeForSubmitter(AssignmentSubmission submission, String submi
         String gradebookAssignmentName = assignment.getProperties().get(PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT);
         if (StringUtils.isNotBlank(gradebookAssignmentName) && !gradebookExternalAssessmentService.isExternalAssignmentDefined(assignment.getContext(), gradebookAssignmentName)) {
             // associated gradebook item
-            grade = gradebookService.getAssignmentScoreStringByNameOrId(assignment.getContext(), gradebookAssignmentName, submitter);
+            grade = gradebookService.getAssignmentScoreString(assignment.getContext(), gradebookAssignmentName, submitter);
         }
```

Comment:
What's the difference between these two calls?

Reply:
getAssignmentScoreStringByNameOrId checks firstly by name and then, if score is null (which can be assignment not found but it is also the default case before grading!) then it checks by id. If it ends up not matching any assignment, it throws an exception. There are other similar deprecated getByName methods on GradebookServiceHibernateImpl that could be used in that process, but I understand why the developer didn't want to.

I talked a bit to Earle because, as bad as that edu-service logic is, it might be like that for a reason. Until we get rid of all the getByName methods, I guess they have to co-exist. That's why I thought it would make more sense to just sort it out for this specific case, it would have less impact.

Reply:
@ern What do you think?

Reply:
If there is an issue with a service method it should be fixed instead of simply switching to a different method and letting someone else deal with it.

From reading https://jira.sakaiproject.org/browse/SAK-29396 This method is deprecated
org.sakaiproject.component.gradebook.GradebookServiceHibernateImpl#getAssignmentScoreString(java.lang.String, java.lang.String, java.lang.String)

So this approach is still incorrect.